### ISSUE TITLE
catkin_destinations not called ?

### ISSUE THREAD

**Issue by asmodehn** [pyros-dev/catkin_pip#119]
While working on rostful on kinetic, I noticed this behavior : 
```
CMake Error at /opt/ros/kinetic/share/catkin/cmake/test/gtest.cmake:128 (_install):
  install DIRECTORY given no DESTINATION!
Call Stack (most recent call first):
  CMakeLists.txt:24 (install)
```

It seems it is caused by catkin variables `CATKIN_PACKAGE_BIN_DESTINATION` and `CATKIN_PACKAGE_SHARE_DESTINATION` being empty.

this would be the case if `catkin_destinations` was never called. But it is supposedly called by `catkin_package_xml()`, called by `catkin_package()`, which we call when the user uses `catkin_pip_package()` https://github.com/asmodehn/catkin_pip/blob/devel/cmake/catkin-pip-package.cmake.in#L121

So all this should work, but in one of my usecase (where I need to use install command) it doesn't...

**Comment by asmodehn** [comment]
This has  been reproduced when working on pyros-rosinterface, trying to install a new launch file.
Still same kinetic + catkin_pip v0.2.0 setup.

**Comment by asmodehn** [comment]
This has been reproduced again with catkin 0.2.2 while working on [ros1_template](https://github.com/pyros-dev/ros1_template)

**Comment by asmodehn** [comment]
The problem seems to be that we are calling the **macro** `catkin_package` from inside a **function** `catkin_pip_package`, and therefore variables are not set in parent scope... fix on the way in #144

**Comment by asmodehn** [comment]
fixed by making `catkin_pip_package` a macro
